guava/src/main/java/com/fasterxml/jackson/datatype/guava/ser/CacheSerializer.java

```diff
@@ -1,41 +1,274 @@
 package com.fasterxml.jackson.datatype.guava.ser;
 
-import java.io.IOException;
-
-import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.core.JsonToken;
 import com.fasterxml.jackson.core.type.WritableTypeId;
-import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.introspect.AnnotatedMember;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonArrayFormatVisitor;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonFormatVisitable;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonFormatVisitorWrapper;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonMapFormatVisitor;
 import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
-import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.databind.ser.ContainerSerializer;
+import com.fasterxml.jackson.databind.ser.ContextualSerializer;
+import com.fasterxml.jackson.databind.ser.PropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.PropertySerializerMap;
 
+import com.fasterxml.jackson.databind.ser.std.MapProperty;
+import com.fasterxml.jackson.databind.type.MapLikeType;
 import com.google.common.cache.Cache;
 
-public class CacheSerializer extends StdSerializer<Cache<?, ?>>
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Serializer for Guava's {@link Cache} values. 
+ * 
+ * @since 2.16
+ */
+public class CacheSerializer 
+    extends ContainerSerializer<Cache<?, ?>>
+    implements ContextualSerializer
 {
     private static final long serialVersionUID = 1L;
 
-    public CacheSerializer() {
-        super(Cache.class, false);
+    private final MapLikeType _type;
+    private final BeanProperty _property;
+    private final JsonSerializer<Object> _keySerializer;
+    private final TypeSerializer _valueTypeSerializer;
+    private final JsonSerializer<Object> _valueSerializer;
+
+    /**
+     * Set of entries to omit during serialization, if any
+     *
+     * @since 2.16
```

Comment:
Since class has `@since 2.16` no need to add it to fields, methods (except later on when adding after 2.16)

Reply:
@JooHyukKim It's ok for now; those are harmless. I can remove them after merging so don't worry about them. Mentioned for future reference. Thanks!

Reply:
Oh. And I was wrong too. This was not added in 2.16 but pre-existing. So new methods should have it. But class not. I'll remove that one (not sure when it was added)